### ISSUE TITLE
False positive 

### ISSUE THREAD

**Issue by tushar-deepsource** [PyCQA/pylint#7735]
### Bug description

```python
match foo:
    case "foo":
        from app.foo import bar
        print("foo")
    case "api":
        import something_else
        from app.api import app
```

### Command used

```shell
pylint asd.py | grep C0412
```

### Pylint output

```shell
asd.py:7:8: C0412: Imports from package app are not grouped (ungrouped-imports)
```

### Expected behavior

No issue raised

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.10.6 (main, Aug 30 2022, 04:58:14) [Clang 13.1.6 (clang-1316.0.21.2.5)]
```

### OS / Environment

MacOS Monterey

**Comment by orSolocate** [comment]
Does re-ordering the import line solves the issue? see [this SO issue](https://stackoverflow.com/questions/45987473/pylint-ungrouped-imports-warning).

```
    case "api":
        from app.api import app
        import something_else
        # do something with the app
```

**Comment by tushar-deepsource** [comment]
@orSolocate even if it does, there's always this prblem:

**Comment by Pierre-Sassoulas** [comment]
As the example where there is an actual false positive is very contrived I'm staring to think this is one case where a disable is warranted. We're going to have a lot of false negative otherwise. Waiting for other maintainers opinion.

**Comment by tusharsadhwani** [comment]
@Pierre-Sassoulas I tried thinking of other reasons for wanting to import inside a match-case, and couldn't think of any.

What I mean is, it's very similar to having an import inside a function definition: you only want that block to run if the case matches.

The imports inside each match case will be completely independent of each other, just like a function definition.

Since Pylint doesn't ask you to group imports inside a function def to the ones outside, why should a match-case be any different? Any false negatives will be correct, imo.
